{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: numpy in c:\\tools\\anaconda3\\lib\\site-packages (1.16.4)\n",
      "Requirement already satisfied: biopython in c:\\tools\\anaconda3\\lib\\site-packages (1.74)\n",
      "Requirement already satisfied: numpy in c:\\tools\\anaconda3\\lib\\site-packages (from biopython) (1.16.4)\n",
      "Requirement already satisfied: pandas in c:\\tools\\anaconda3\\lib\\site-packages (0.24.2)\n",
      "Requirement already satisfied: numpy>=1.12.0 in c:\\tools\\anaconda3\\lib\\site-packages (from pandas) (1.16.4)\n",
      "Requirement already satisfied: pytz>=2011k in c:\\tools\\anaconda3\\lib\\site-packages (from pandas) (2019.1)\n",
      "Requirement already satisfied: python-dateutil>=2.5.0 in c:\\tools\\anaconda3\\lib\\site-packages (from pandas) (2.8.0)\n",
      "Requirement already satisfied: six>=1.5 in c:\\tools\\anaconda3\\lib\\site-packages (from python-dateutil>=2.5.0->pandas) (1.12.0)\n",
      "Requirement already satisfied: scikit-learn in c:\\tools\\anaconda3\\lib\\site-packages (0.21.2)\n",
      "Requirement already satisfied: scipy>=0.17.0 in c:\\tools\\anaconda3\\lib\\site-packages (from scikit-learn) (1.2.1)\n",
      "Requirement already satisfied: joblib>=0.11 in c:\\tools\\anaconda3\\lib\\site-packages (from scikit-learn) (0.13.2)\n",
      "Requirement already satisfied: numpy>=1.11.0 in c:\\tools\\anaconda3\\lib\\site-packages (from scikit-learn) (1.16.4)\n"
     ]
    }
   ],
   "source": [
    "import sys\n",
    "!{sys.executable} -m pip install numpy\n",
    "!{sys.executable} -m pip install biopython\n",
    "!{sys.executable} -m pip install pandas\n",
    "!{sys.executable} -m pip install scikit-learn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\tools\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py:3057: DtypeWarning: Columns (6,8,9,10,11,12) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "import sys\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "\n",
    "cancer_genes = pd.read_csv(\"C:\\\\Users\\\\monju\\\\OneDrive\\\\bioinformatics-data\\\\cancer_genes_latest.csv\")\n",
    "cell_line_details = pd.read_excel(\"C:\\\\Users\\\\monju\\\\OneDrive\\\\bioinformatics-data\\\\cell_line_details.xlsx\")\n",
    "cosmic_tissue_classification = pd.read_excel(\"C:\\\\Users\\\\monju\\\\OneDrive\\\\bioinformatics-data\\\\cosmic_tissue_classification.xlsx\")\n",
    "gdsc1 = pd.read_excel(\"C:\\\\Users\\\\monju\\\\OneDrive\\\\bioinformatics-data\\\\gdsc1_fitted_dose_response.xlsx\")\n",
    "gdsc2 = pd.read_excel(\"C:\\\\Users\\\\monju\\\\OneDrive\\\\bioinformatics-data\\\\gdsc2_fitted_dose_response.xlsx\")\n",
    "gene_identifiers_modified = pd.read_csv(\"C:\\\\Users\\\\monju\\\\OneDrive\\\\bioinformatics-data\\\\gene_identifiers_modified.csv\")\n",
    "model_list = pd.read_csv(\"C:\\\\Users\\\\monju\\\\OneDrive\\\\bioinformatics-data\\\\model_list_latest.csv\")\n",
    "mutations_latest = pd.read_csv(\"C:\\\\Users\\\\monju\\\\OneDrive\\\\bioinformatics-data\\\\mutations_latest.csv\")\n",
    "screened_compounds = pd.read_csv(\"C:\\\\Users\\\\monju\\\\OneDrive\\\\bioinformatics-data\\\\screened_compounds_rel_8.0.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ABCB1\n",
      "ABL2\n",
      "ACACA\n",
      "ACVR1B\n",
      "ACVR2A\n",
      "ADCY1\n",
      "ADGRL2\n",
      "AFDN\n",
      "AFF4\n",
      "AHCTF1\n",
      "AHNAK\n",
      "AKAP9\n",
      "AKT1\n",
      "ALK\n",
      "AMER1\n",
      "AMOT\n",
      "ANK3\n",
      "APAF1\n",
      "APC\n",
      "AQR\n",
      "ARAP3\n",
      "ARFGAP3\n",
      "ARFGEF1\n",
      "ARFGEF2\n",
      "ARHGAP26\n",
      "ARHGAP29\n",
      "ARHGEF6\n",
      "ARID1A\n",
      "ARID2\n",
      "ARID4A\n",
      "ARID4B\n",
      "ARID5B\n",
      "ASH1L\n",
      "ASPM\n",
      "ASXL1\n",
      "ATM\n",
      "ATR\n",
      "ATRX\n",
      "AXIN1\n",
      "AXIN2\n",
      "B2M\n",
      "BAP1\n",
      "BAZ2B\n",
      "BCLAF1\n",
      "BCOR\n",
      "BLM\n",
      "BMPR2\n",
      "BNC2\n",
      "BPTF\n",
      "BRAF\n",
      "BRCA1\n",
      "BRCA2\n",
      "BRWD1\n",
      "CAD\n",
      "CASP8\n",
      "CCAR1\n",
      "CCND1\n",
      "CDC27\n",
      "CDC73\n",
      "CDH1\n",
      "CDK12\n",
      "CDKN1A\n",
      "CDKN1B\n",
      "CDKN2A\n",
      "CEP290\n",
      "CHD4\n",
      "CHD6\n",
      "CHD9\n",
      "CIC\n",
      "CLASP2\n",
      "CLOCK\n",
      "CLSPN\n",
      "CMTR2\n",
      "CNOT1\n",
      "CNOT3\n",
      "COL1A1\n",
      "CREBBP\n",
      "CTCF\n",
      "CTNNB1\n",
      "CTNND1\n",
      "CUL2\n",
      "CUL3\n",
      "CUX1\n",
      "CYLD\n",
      "DDX3X\n",
      "DDX5\n",
      "DHX15\n",
      "DHX9\n",
      "DICER1\n",
      "DLG1\n",
      "DNMT3A\n",
      "EGFR\n",
      "EIF4G1\n",
      "ELF1\n",
      "ELF3\n",
      "EPHA2\n",
      "EPHA4\n",
      "ERBB2\n",
      "ERBB3\n",
      "ERBIN\n",
      "EZH2\n",
      "F8\n",
      "FAF1\n",
      "FAT1\n",
      "FAT2\n",
      "FBXW7\n",
      "FIP1L1\n",
      "FLT3\n",
      "FN1\n",
      "FOXP1\n",
      "FRG1BP\n",
      "FXR1\n",
      "G3BP2\n",
      "GATA3\n",
      "GNA11\n",
      "GPS2\n",
      "HDAC9\n",
      "HGF\n",
      "HLA-A\n",
      "HLA-B\n",
      "HNF1A\n",
      "HRAS\n",
      "HSPA8\n",
      "IDH1\n",
      "IDH2\n",
      "INPP4A\n",
      "INPPL1\n",
      "IREB2\n",
      "ITSN1\n",
      "KALRN\n",
      "KAT6B\n",
      "KDM5C\n",
      "KDM6A\n",
      "KDR\n",
      "KEAP1\n",
      "KIT\n",
      "KMT2A\n",
      "KMT2D\n",
      "KRAS\n",
      "LAMA2\n",
      "LARP4B\n",
      "LIMA1\n",
      "LNPEP\n",
      "LRP6\n",
      "LRPPRC\n",
      "MACF1\n",
      "MAGI2\n",
      "MAP2K1\n",
      "MAP2K4\n",
      "MAP3K1\n",
      "MAP3K4\n",
      "MAP4K3\n",
      "MAX\n",
      "MCM3\n",
      "MECOM\n",
      "MED12\n",
      "MED23\n",
      "MED24\n",
      "MET\n",
      "MGA\n",
      "MKL1\n",
      "MLH1\n",
      "MLH3\n",
      "MNDA\n",
      "MSR1\n",
      "MTOR\n",
      "MYC\n",
      "MYCN\n",
      "MYH11\n",
      "MYH14\n",
      "MYH9\n",
      "NCOR1\n",
      "NCOR2\n",
      "NF1\n",
      "NF2\n",
      "NFE2L2\n",
      "NOTCH1\n",
      "NOTCH2\n",
      "NPM1\n",
      "NR4A2\n",
      "NRAS\n",
      "NSD1\n",
      "NSD2\n",
      "NSD3\n",
      "NTN4\n",
      "NTRK2\n",
      "NUP93\n",
      "NUP98\n",
      "NUTM1\n",
      "PABPC3\n",
      "PBRM1\n",
      "PCDH18\n",
      "PCSK5\n",
      "PGR\n",
      "PHF6\n",
      "PHLPP1\n",
      "PIK3C2B\n",
      "PIK3CA\n",
      "PIK3CB\n",
      "PIK3R1\n",
      "PLCB1\n",
      "PLCG1\n",
      "PLXNA1\n",
      "POLR2B\n",
      "PPP2R1A\n",
      "PSIP1\n",
      "PTCH1\n",
      "PTEN\n",
      "PTPN11\n",
      "PTPRU\n",
      "RAC1\n",
      "RAD21\n",
      "RASA1\n",
      "RASA2\n",
      "RB1\n",
      "RBM10\n",
      "RBM5\n",
      "RGS3\n",
      "RHOA\n",
      "RHOT1\n",
      "RNF43\n",
      "ROBO2\n",
      "RPGR\n",
      "RTN4\n",
      "RUNX1\n",
      "SCAI\n",
      "SEC31A\n",
      "SETD2\n",
      "SETDB1\n",
      "SF3B1\n",
      "SIN3A\n",
      "SMAD2\n",
      "SMAD4\n",
      "SMARCA1\n",
      "SMARCA4\n",
      "SMARCB1\n",
      "SMC1A\n",
      "SOS1\n",
      "SOS2\n",
      "SOX9\n",
      "SPOP\n",
      "SPTAN1\n",
      "SRGAP1\n",
      "SRGAP3\n",
      "STAG1\n",
      "STAG2\n",
      "STIP1\n",
      "STK11\n",
      "STK4\n",
      "SVEP1\n",
      "TAOK1\n",
      "TBL1XR1\n",
      "TBX3\n",
      "TCF12\n",
      "TCF4\n",
      "TCF7L2\n",
      "TET2\n",
      "THRAP3\n",
      "TJP1\n",
      "TJP2\n",
      "TP53\n",
      "TRIO\n",
      "TSC1\n",
      "U2AF1\n",
      "UPF3B\n",
      "USP6\n",
      "VHL\n",
      "WNK1\n",
      "XPO1\n",
      "XRN1\n",
      "ZC3H11A\n",
      "ZFHX3\n",
      "ZFP36L1\n",
      "ZFP36L2\n",
      "ZMYM2\n",
      "ZNF292\n",
      "ZNF638\n",
      "ZNRF3\n"
     ]
    }
   ],
   "source": [
    "from Bio import Entrez\n",
    "\n",
    "cancer_gene_sequences = {};\n",
    "Entrez.email=\"monjurmorshed793@gmail.com\"\n",
    "counter = 0;\n",
    "for index, row in cancer_genes.iterrows():\n",
    "    counter +=1\n",
    "    print(row[0])\n",
    "    handle = Entrez.esearch(db='gene', term=row[0])\n",
    "    record = Entrez.read(handle)\n",
    "    for i in record['IdList']:\n",
    "        executed = False\n",
    "        try:\n",
    "            gene_data = Entrez.efetch(db=\"nucleotide\", id=i, rettype=\"fasta\", retmode=\"fasta\")\n",
    "#             cancer_gene_sequences.append(gene_data.read)\n",
    "            sequence = SeqIO.parse(gene_data, 'fasta')\n",
    "            cancer_gene_sequences[row[0]]=str(gene_data.read())\n",
    "            sequence_seq = next(sequence)\n",
    "            gene_data_str = str(sequence_seq.seq)\n",
    "            print('------')\n",
    "            print(gene_data_str)\n",
    "            if 'NNN' in gene_data_str:\n",
    "                executed = False\n",
    "            else:\n",
    "                executed=True\n",
    "#             f = open(\"C:\\\\Users\\\\monju\\\\Documents\\\\bioinformatics-data\\\\cancer_genes\\\\\"+row[0]+\".fasta\",\"w\")\n",
    "#             f.write(next(sequence))\n",
    "#             f.close()\n",
    "        except:\n",
    "            executed=False    \n",
    "        if executed==True:\n",
    "            break\n",
    "  "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
